### PR TITLE
Attempting to fix issues #3 and #4

### LINKED ISSUE #3: Bug: Cannot read property 'duration_ms' of undefined
From line: https://github.com/bitfocus/companion-module-spotify-remote/blob/0fdb9599bb08e4b81c0d351f000b974554b3fe5a/module.js#L221

This error has been reported in sentry over 152k times in 2 days

### PR DESCRIPTION
**Pull request by Simber1**
This should either fix #3 and #4 or will at least give me useful errors to work with. Could @Julusian test this please to see if the issues still exist.

### CODE REVIEW

File: module.js [hunk 1]
```diff
@@ -217,19 +211,25 @@ instance.prototype.GetPlaybackState = function(){
...
-        
-        songPercentage = songPercentage*100;
-        songPercentage = songPercentage.toFixed(0);
-
-        songProgress = songProgress/1000;
-        songProgress = songProgress.toFixed(0);
-
-        songDuration = songDuration/1000;
-        songDuration = songDuration.toFixed(0);
+        if (data.body.item.duration_ms) {
```
Comment (Julusian) [inline module.js:214]: This should be `if (data.body.item) {` as the problem was it was undefined causing trying to read the `.duration_ms` property to throw an error

File: module.js [hunk 2]
```diff
@@ -217,19 +211,25 @@ instance.prototype.GetPlaybackState = function(){
...
+            songPercentage = songPercentage*100;
+            songPercentage = songPercentage.toFixed(0);
+    
+            songProgress = songProgress/1000;
+            songProgress = songProgress.toFixed(0);
+    
+            songDuration = songDuration/1000;
+            songDuration = songDuration.toFixed(0);
+        } else {
+            var songProgress = 0;
```
Comment (Julusian) [inline module.js:228]: you are going to have a scope issue here. these variables will only exist inside this else block, and will be undefined by the time you are calling `self.setVariable(...` with them

**Review by Julusian — CHANGES_REQUESTED** [review]

File: module.js (hunk 1 above)
Comment (Simber1) [reply module.js]: Fixed this.

File: module.js (hunk 2 above)
Comment (Simber1) [reply module.js]: Fixed this by moving the variable definitions outside of the loop, also noticed that some other variables would have had the same issue and gave them defaults for when data.body.item is empty.

**Review by Julusian — APPROVED** [review]
Looking good now.
I gave it a quick test and managed to get another error state, but I pushed a fix for that.
This is good to be merged now

**Comment by Simber1** [comment]
I've added a check for if the device is a mobile or tablet since they can't have their volumes changed via the API. There might be other devices that can't be changed like a cast recipient but I can't test them right now.

**Pull request merged**